OpenRA.Mods.Common/HitShapes/IHitShape.cs

```diff
@@ -0,0 +1,24 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.HitShapes
+{
+	public interface IHitShape
+	{
+		WDist OuterRadius { get; }
+
+		WDist DistanceFromEdge(WVec p);
```

Comment:
This requires `p` to be already rotated into the local coordinate system of the victim (see SpreadDamageWarhead:L69) which breaks the abstraction a bit, IMO.

How about we make this `WDist DistanceFromEdge(Actor self, WPos position)`, and then calculate the vector, rotate it, and return the distance all internally.  The circle shape can then just return the vector distance directly.